### PR TITLE
Add save to Prisma

### PR DESCRIPTION
**Pull request by shuding**

### CODE REVIEW

**Review by jaredpalmer — APPROVED** [review]

**Review by kwasham — COMMENTED** [review]
VERDICT: PASS

Security review summary:
- Authentication now flows through `lib/auth`: Clerk sessions are validated on every route, middleware protects non-public paths, and guest sessions are still backed by httpOnly, secure cookies.
- `/api/auth/guest` now rejects any redirect that does not begin with `/`, so the open redirect vulnerability is closed.
- Environment variables for Clerk are referenced only in `.env.example` with placeholder values—no credentials leaked.
- CI status: no checks reported for this branch (please exercise the normal pipeline if needed).

**Review by kwasham — CHANGES_REQUESTED** [review]
FAIL: Reviewed the Clerk migration, and the guest endpoint now enforces that  starts with  and never with , so the open-redirect fix is in place and guest cookies remain httpOnly/secure. However, the Vercel deployment is currently failing (https://vercel.com/kwashams-projects/vercel-ai-chatbot/7iKMHgBdwgnFRsi37Mjo4pkxufJw — ), so the PR cannot be merged while CI is red. Please fix the failing deployment (e.g., supply the required Clerk environment variables or resolve the build error) and rerun CI before I can approve.

**Review by kwasham — CHANGES_REQUESTED** [review]
FAIL: Reviewed the Clerk migration and guest endpoint. The new `lib/auth` returns Clerk-managed sessions with a guest fallback, and `/api/auth/guest` now enforces that redirectUrl starts with "/" and never //, so the previous open-redirect class is fixed while the guest cookie stays httpOnly/secure. However, the Vercel deployment is still failing (https://vercel.com/kwashams-projects/vercel-ai-chatbot/7iKMHgBdwgnFRsi37Mjo4pkxufJw — npx vercel inspect dpl_7iKMHgBdwgnFRsi37Mjo4pkxufJw --logs), so the CI barrier is red and this cannot be merged until the deployment succeeds. Please resolve the failing build and rerun the checks before I can approve.

**Pull request merged**
